config/src/test/kotlin/org/springframework/security/config/web/server/ServerHeadersDslTests.kt

```diff
@@ -198,4 +199,50 @@ class ServerHeadersDslTests {
             }
         }
     }
+
+    @Test
+    fun `request when custom server http headers writer configured then custom http headers added`() {
+        this.spring.register(ServerHttpHeadersWriterCustomConfig::class.java).autowire()
+
+        this.client.get()
+            .uri("/")
+            .exchange()
+            .expectHeader().valueEquals("CUSTOM-HEADER-1", "CUSTOM-VALUE-1")
+            .expectHeader().valueEquals("CUSTOM-HEADER-2", "CUSTOM-VALUE-2")
+    }
+
+    @Configuration
+    @EnableWebFluxSecurity
+    @EnableWebFlux
+    open class ServerHttpHeadersWriterCustomConfig {
+        @Bean
+        open fun springWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
+            return http {
+                headers {
+                    serverHttpHeadersWriter {
```

Comment:
While this looks nice on the Kotlin side, it's a little clumsy on the Java side. What about this instead:

```
headers {
    writer { exchange ->
    writer { exchange ->
}
```

I think this still looks quite nice on the Kotlin side and doesn't add another DSL layer on the Java side.

Reply:
Done